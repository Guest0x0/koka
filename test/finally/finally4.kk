// see issue #360
effect ctl myeffect() : ()

fun main() : <console> ()
    with finally
      println("top")
    with ctl myeffect()  ()
    with finally
      println("mid")
    with ctl myeffect()  myeffect()
    with finally
      println("cleanup")
    myeffect()