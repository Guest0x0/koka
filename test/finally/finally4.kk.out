cleanup
mid
top
 
finally/finally4/main: () -> console ()
finally/finally4/myeffect: () -> myeffect ()